internal/contour/holdoff.go

```diff
@@ -42,16 +42,36 @@ type HoldoffNotifier struct {
 	mu    sync.Mutex
 	timer *time.Timer
 	last  time.Time
+
+	holdoffDelay    time.Duration
+	holdoffMaxDelay time.Duration
+}
+
+func (hn *HoldoffNotifier) SetHoldoffDelay(t time.Duration) {
+	hn.holdoffDelay = t
+}
+
+func (hn *HoldoffNotifier) SetHoldoffMaxDelay(t time.Duration) {
+	hn.holdoffMaxDelay = t
 }
 
 func (hn *HoldoffNotifier) OnChange(builder *dag.Builder) {
 	hn.mu.Lock()
 	defer hn.mu.Unlock()
+
+	if hn.holdoffDelay == 0 {
```

Comment:
This check is only needed once, I was thinking of removing this check by using a constructor func but this seemed cleaner.

Reply:
What happens if someone calls `SetHoldoffDelay(0)`, their value will be overridden with the default.